### PR TITLE
Add class to fetch organizations

### PR DESCRIPTION
**Pull request by dimasciput**
Issue #21

### CODE REVIEW

File: cadasta/api/organization.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+class Organization(object):
+    """Organization api class"""
+
+    api_url = 'https://demo.cadasta.org/api/v1/organizations/'
+
+    def get_api(self, network):
+        """
+        Execute api get call
+        :param network: NetworkMixin object
```
Comment (timlinux) [inline cadasta/api/organization.py:29]: Its good to use :type network: annotations too - pychar will do at-code-time type checking for you if you do.

File: cadasta/api/organization.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from cadasta.mixin.network_mixin import NetworkMixin
+
+__author__ = 'dimas@kartoza.com'
+__revision__ = '$Format:%H$'
+__date__ = '26/02/2014'
+__copyright__ = 'Copyright 2016, Cadasta'
+
+
+class Organization(object):
+    """Organization api class"""
```
Comment (timlinux) [inline cadasta/api/organization.py]: To be a little pedantic, your first line of your docstring should be terminated by a full-stop and then followed by a newline. Sphinx and docstring will use that to format the __help__ etc text nicely. Same applies for first line of function docstrings. Please read this section carefully for more background: https://github.com/inasafe/inasafe/wiki/Coding-Standards#doc-strings-and-comments

File: cadasta/api/organization.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        """
+        network.connect_get()
+        while not network.reply.isFinished():
+            QCoreApplication.processEvents()
+        if not network.error:
+            return True, network.get_json_results()
+        else:
+            return False, network.error
+
+    def get_all_organizations(self):
```
Comment (timlinux) [inline cadasta/api/organization.py]: I think generally get_ prefix is superfluous and makes code more verbose. Suggest to remove them all...

File: cadasta/api/test/test_organization.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    def setUp(self):
+        """Runs before each test."""
+        self.username = 'kartoza.demo'
+        self.password = 'demo.kartoza1!'
+
+    def tearDown(self):
+        """Runs after each test."""
+        self.dialog = None
+
+    def test_get_all_organizations(self):
```
Comment (timlinux) [inline cadasta/api/test/test_organization.py:35]: Note that I would suggest that you consider using mock object for these tests. The mock object can return a canned response from the real service. If we do that we can avoid network overhead while testing and also have a deterministic result.

Comment (timlinux) [reply cadasta/api/test/test_organization.py:35]: Also we dont then need to store real credentials in our code....

File: cadasta/api/organization.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class Organization(object):
+    """Organization api class"""
+
+    api_url = 'https://demo.cadasta.org/api/v1/organizations/'
+
+    def get_api(self, network):
+        """
+        Execute api get call
```
Comment (Gustry) [inline cadasta/api/organization.py]: Please follow this link for the docstring : https://github.com/inasafe/inasafe/wiki/Coding-Standards#doc-strings-and-comments.
The example is correct.
* A full sentence with a dot.
* New line between parameters
* Add `:rtype`

File: cadasta/api/test/test_organization.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
```
Comment (ismailsunni) [inline cadasta/api/test/test_organization.py:1]: Please add docstring for the file.

**Review by ismailsunni — APPROVED** [review]
Small comment, LGTM

**Pull request merged**